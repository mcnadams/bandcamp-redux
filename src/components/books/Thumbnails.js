import React from 'react';
import PropTypes from 'prop-types';
import Thumbnail from './Thumbnail';

function Thumbnails({ isbns }) {
  const bookList = isbns.map(isbn => {
    return (
      <li key={isbn}>
        <Thumbnail isbn={isbn} />
      </li>
    );
  });

  return (
    <ul>{bookList}</ul>
  );
}


Thumbnails.propTypes = {
  isbns: PropTypes.array.isRequired
};

export default Thumbnails;
