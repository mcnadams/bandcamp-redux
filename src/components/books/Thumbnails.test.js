import { shallow } from 'enzyme';
import React from 'react';
import Thumbnails from './Thumbnails';

describe('Thumbnail component', () => {
  it('renders thumbnail', () => {
    const wrapper = shallow(<Thumbnails isbns={['222', '333']}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
