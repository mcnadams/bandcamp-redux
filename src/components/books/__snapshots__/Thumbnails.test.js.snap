// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail component renders thumbnail 1`] = `
<ul>
  <li
    key="222"
  >
    <Thumbnail
      isbn="222"
    />
  </li>
  <li
    key="333"
  >
    <Thumbnail
      isbn="333"
    />
  </li>
</ul>
`;
